microservices/netinfra_utils/workers/netconf_worker_test.py

```diff
@@ -0,0 +1,142 @@
+    def test_unmount_existing_device(self):
+        with patch('netconf_worker.requests.delete') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps({}), encoding='utf-8'), 204)
+            request = netconf_worker.execute_unmount_netconf({"inputData": {"device_id": "xr6"}})
+            self.assertEqual(request["status"], "COMPLETED")
+            self.assertEqual(request["output"]["url"], odl_url_base
+                             + "/data/network-topology:network-topology/topology=topology-netconf/node=xr6")
+            self.assertEqual(request["output"]["response_code"], 204)
+            self.assertEqual(request["output"]["response_body"], {})
+
+
+class TestCheckCliConnected(unittest.TestCase):
+    def test_execute_check_connected_netconf_connecting(self):
+        with patch('netconf_worker.requests.get') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps(netconf_node_connecting), encoding='utf-8'), 200)
+            request = netconf_worker.execute_check_connected_netconf({"inputData": {"device_id": "xr6"}})
+            self.assertEqual(request["status"], "FAILED")
+            self.assertEqual(request["output"]["url"], odl_url_base
+                             + "/data/network-topology:network-topology/topology=topology-netconf"
+                               "/node=xr6?content=nonconfig")
+            self.assertEqual(request["output"]["response_body"]["node"][0]["netconf-node-topology:connection-status"],
+                             "connecting")
+
+    def test_execute_check_connected_netconf_connected(self):
+        with patch('netconf_worker.requests.get') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps(netconf_node_connected), encoding='utf-8'), 200)
+            request = netconf_worker.execute_check_connected_netconf({"inputData": {"device_id": "xr6"}})
+            self.assertEqual(request["status"], "COMPLETED")
+            self.assertEqual(request["output"]["url"], odl_url_base
+                             + "/data/network-topology:network-topology/topology=topology-netconf"
+                               "/node=xr6?content=nonconfig")
+            self.assertEqual(request["output"]["response_body"]["node"][0]["netconf-node-topology:connection-status"],
+                             "connected")
+
+
+class TestCheckNetconfIdAvailable(unittest.TestCase):
+    def test_execute_check_netconf_id_available_exist(self):
+        with patch('netconf_worker.requests.get') as mock:
+            mock.return_value = MockResponse(bytes(json.dumps(netconf_node_connected), encoding='utf-8'), 200)
+            request = netconf_worker.execute_check_netconf_id_available({"inputData": {"device_id": "xr6"}})
+            self.assertEqual(request["status"], "FAILED")
+            self.assertEqual(request["output"]["url"], odl_url_base
+                             + "/data/network-topology:network-topology/topology=topology-netconf/node=xr6")
```

Comment:
request URL should contain ?content=config

Reply:
fixed